editor/src/clj/editor/buffers.clj

```diff
@@ -208,6 +349,79 @@
           byte-order-comparison
           (compare (total-byte-size a) (total-byte-size b)))))))
 
+;; A wrapper around Buffer with equality semantics based on the topology of the
+;; Buffer and an explicit data version number.
+(defonce/type BufferData
+  [^Buffer data
+   ^int data-version
+   ^int topology-hash]
```

Comment:
We use this wrapper around `Buffer` to avoid expensive equality tests when you need to supply a buffer as part of the `request-data` provided to `scene-cache` requests. The `Buffer` implementations do proper content equality, bless them, but we'd rather manage equality ourselves by comparing a version number.